### PR TITLE
PLT-1917 Add 'new messages' indicator when scrolled up

### PR DESCRIPTION
**Pull request by alanmoo**
I used the existing i18n key for quicker integration, but can create a new one if that's really desired.

### CODE REVIEW

**Comment by hmhealey** [comment]
Hey Alan, you've got some ESLint failures preventing the code from building. You can run that on your local instance by running `make test` in the webapp directory.
```
> eslint --ext ".jsx" --ignore-pattern node_modules --quiet .


/var/lib/jenkins/jobs/mattermost-platform-pr/workspace/src/github.com/mattermost/platform/webapp/components/posts_view.jsx
  426:36  error  'nextState' is defined but never used                                                           no-unused-vars
  427:8   error  Expected indentation of 8 space characters but found 7                                          indent
  428:47  error  Trailing spaces not allowed                                                                     no-trailing-spaces
  429:9   error  Expected indentation of 7 space characters but found 8                                          indent
  520:21  error  Trailing spaces not allowed                                                                     no-trailing-spaces
  521:31  error  Unexpected usage of doublequote                                                                 jsx-quotes
  523:58  error  The closing bracket must be aligned with the opening tag (expected column 17 on the next line)  react/jsx-closing-bracket-location
  524:38  error  Unexpected usage of doublequote                                                                 jsx-quotes
  527:33  error  Expected indentation of 28 space characters but found 32                                        react/jsx-indent-props
  528:33  error  Expected indentation of 28 space characters but found 32                                        react/jsx-indent-props
  529:29  error  The closing bracket must be aligned with the opening tag (expected column 25)                   react/jsx-closing-bracket-location
```

**Comment by alanmoo** [comment]
Ah, I always forget to run that and my new editor isn't autocorrecting properly. Fixed.

**Comment by lfbrock** [comment]
Hi @alanmoo, I took a look this morning and it looked good - was just going to test more with the team in UX but we're all getting console errors now (the centre channel just shows a continuous loading message).

**Comment by alanmoo** [comment]
Ah, I think I know what's going on- forgot about the (very much non-edge) case of empty channels. I'll fix that now

**Comment by alanmoo** [comment]
@lfbrock give it a shot now

**Comment by alanmoo** [comment]
Is this going to get stuck in feature freeze?

**Comment by it33** [comment]
Hi @alanmoo, 

Sorry, PRs on master won't be out until May 16 release. 

Last week we announced that v2.2 shipping on April 16 is a special build off of the v2.1 branch (introducing French language and some minor fixes), and that [master ships May 16](https://forum.mattermost.org/t/fyi-updates-on-v2-2-for-april-16-and-v2-3-for-may-16/1201), because of the scale of some of the larger changes going in. 

You've brought up the issue of communication a few times and I agree we need to get better, going to see if we can pipe [announcements](https://forum.mattermost.org/c/announce) on forums into the Reception channel so it's easier to monitor...

**Comment by lfbrock** [comment]
Hi @alanmoo, tested and found a couple cases with empty channels that had a bug, but looked good other than that:

**Case 1:**
1. Create a new channel
2. Add a member

Observed: "New messages" indicator shows up at bottom of channel and doesn’t disappear

**Case 2:**
1. User A opens a new direct message channel
2. User B has the direct messages channel open as well
3. User A sends a message

Observed: User B sees the “New Messages” indicator stuck to the bottom of the page

**Comment by alanmoo** [comment]
@lfbrock Ok, cool, I think I know where that's coming from.

@it33 Ah, ok, that makes sense. A personal experience comment with regards to the reception channel though: I've turned off notifications for that one entirely, as the signal to noise is not high enough for what I'm looking for. Now that I realize this info is in the announcements forum, I can keep an eye on that. It's just hard to know where to go as you have so many sources of info. (Is there an RSS feed I can subscribe to?)

**Comment by it33** [comment]
Not sure about RSS, maybe join the [Community Heartbeat](https://pre-release.mattermost.com/core/channels/community-heartbeat) channel, set notifications to mention-only and add a keyword notification for "[Announcements]"?

**Comment by alanmoo** [comment]
@lfbrock Fixed both of those

File: webapp/components/posts_view.jsx [hunk 1]
```diff
@@ -425,6 +425,15 @@ export default class PostsView extends React.Component {
 
         return false;
     }
+    componentWillUpdate(nextProps) {
+        if (this.props.postList && this.props.postList.order.length) {
```
Comment (hmhealey) [inline webapp/components/posts_view.jsx]: Can you move this into componentWillReceiveProps and move showUnreadMessageAlert into the state? Since it's UI state, it should be part of that and not just another field on the component

Comment (alanmoo) [inline webapp/components/posts_view.jsx]: Good call, I think I once had a reason for it to be in WillUpdate, but I think I'd simplified since then. I also changed the location of one of the `if/else` conditions to cover an edge case of adding a user to a channel and the first message making the alert show.

**Comment by alanmoo** [comment]
Can this get a Setup Test Server label toggle to spin it up again?

**Comment by jwilander** [comment]
+1

**Comment by crspeller** [comment]
+1
